arch/sim/src/sim/up_hostmemory.c

```diff
@@ -140,5 +140,22 @@ void *host_calloc(size_t n, size_t elem_size)
 
 void *host_memalign(size_t alignment, size_t size)
 {
+  /* Note: macOS prior to 10.15 (Catalina) doesn't have aligned_alloc().  */
+
+#if !defined(__APPLE__) || \
+   (defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) && \
+    __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 101500)
   return aligned_alloc(alignment, size);
+#else
```

Comment:
Thanks for fix, but @yamt can we remvoe #if/#else/#endif and always call posix_memalign instead?

Reply:
do all other relevant platforms have posix_memalign?

Reply:
from https://man7.org/linux/man-pages/man3/posix_memalign.3.html:
```
posix_memalign require _POSIX_C_SOURCE >= 200112L
aligned_alloc require _ISOC11_SOURCE
```
so posix_memalign is more safer than aligned_alloc.

Reply:
ok. done.

i also added a patch to fix a build error with CONFIG_SIM_SANITIZE=y.

Reply:
Thanks.